scarb/src/core/registry/client/http.rs

```diff
@@ -146,12 +151,77 @@ impl<'c> RegistryClient for HttpRegistryClient<'c> {
     }
 
     async fn supports_publish(&self) -> Result<bool> {
-        // TODO(mkaput): Publishing to HTTP registries is not implemented yet.
-        Ok(false)
+        let index_config = self.index_config.load().await?;
+        match index_config.upload {
+            Some(_) => Ok(true),
+            None => Ok(false),
+        }
     }
 
-    async fn publish(&self, _package: Package, _tarball: FileLockGuard) -> Result<()> {
-        todo!("Publishing to HTTP registries is not implemented yet.")
+    async fn publish(&self, package: Package, tarball: FileLockGuard) -> Result<RegistryUpload> {
+        let auth_token = env::var("SCARB_REGISTRY_AUTH_TOKEN")
+            .map_err(|_| anyhow!("missing authentication token"))?;
+
+        let path = tarball.path().to_owned();
+        // we need to drop, because windows file locking is very strict
+        drop(tarball);
+
+        ensure!(
+            Path::new(&path).exists(),
+            "cannot upload package - file does not exist at path: {}",
+            &path
+        );
+
+        let file = TokioFile::open(&path).await?;
+        let metadata = file.metadata().await?;
+        ensure!(
+            metadata.len() < 5 * 1024 * 1024,
```

Comment:
why are we hardcoding a seemingly arbitrary number on client side?

Reply:
I can imagine one creating a custom registry with a higher package size limit. In such scenario, we'd be the bad guys for hardcoding some arbitrary limit client-side.

Won't relying on the registry to reject post requests with huge content-size be enough? (+/-)

Reply:
The registry would then probably need to accept the file, in order to check its metadata, right? so the resources would already be wasted. It's not a big deal, but still. 
Looking at the package sizes atm we are comfortably in the < 100Kb region, just for context. I know things may change, but on the other hand I really don't think they will anytime soon, if ever.

Reply:
> And if there’s already a server-side limit, adding the same check client-side feels redundant.

Please correct me if I'm wrong, but I don't think there is such limit atm

> but we're already doing that when checking that:

I think those are quite different cases. 
Moreover I'm pretty sure that checks for missing dependencies are done locally (granted - registry is being queried, but this is not something that registry does when package is uploaded, which is the case with publish). Version duplication or keywords limit is not resource intensive. 

I do understand your overall points though, and I do not disagree with them. We can try and look into file upload validation on django side, to see it we can somehow manage to do that before upload starts (I imagine this is quite common problem, so perhaps a solution alredy exists). I can raise an issue, and remove this check from scarb for time being if you wish so. sounds good?

Reply:
> Please correct me if I'm wrong, but I don't think there is such limit atm

Yes, we don't have it atm 😅 what I meant is that we definitely *should have it* on the registry side, so there will be unnecessary duplication anyway

> I can raise an issue, and remove this check from scarb for time being if you wish so. sounds good?

Sounds good

Reply:
Great. Created an issue: https://github.com/software-mansion/scarbs.xyz/issues/352
and removed said code: https://github.com/software-mansion/scarb/pull/1570/commits/9d900b90d48df046bb6f3d5aa3d658e4b6420352